build_tools/azure/get_commit_message.py

```diff
@@ -19,6 +19,17 @@ def get_commit_message():
     else:
         commit_message = build_source_version_message
 
+    # Sanitize the commit message to avoid introducing a vulnerability: a PR
+    # submitter could include the "##vso" special marker in their commit
+    # message to attempt to obfuscate the injection of arbitrary commands in
+    # the Azure pipeline.
+    #
+    # This can be a problem if the PR reviewers to not pay close enough
+    # attention to the full commit message prior to clicking the merge button
+    # and as a result make the inject code run in a protected branch with
+    # elevated access to CI secrets.
+    commit_message = commit_message.replace("##vso", "..vso")
```

Comment:
Related to this PR: I might have been (one of) the culprit(s) when merging pull requests, integrating unneeded body in the commit message _e.g._ recently at least for:
 - https://github.com/scikit-learn/scikit-learn/issues/25293
 - https://github.com/scikit-learn/scikit-learn/pull/25323

I have two questions in this regard:
 - I do not remember having body integrated when merging PR before. Is GitHub proposing injecting those body in before merging, or is this due to extensions (e.g. Refined GitHub)?
 - Do we want to performed some filtering to remove (parts of) merge commit messages' body?

Reply:
When squashing, both git and github concatenate the body of all squashed commits by default. But the squashing person can interactively edit this if they wish.

Reply:
> Do we want to sanitize (parts of) merge commit messages' body (using more complex patterns)?

I don't think we need to do anything to the merge commit message themselves at merge time. It's just that our CI scripts should always be careful when printing anything on stdout that might be derived from user controllable inputs.

Reply:
> Do we want to sanitize (parts of) merge commit messages' body (using more complex patterns)?

I think the "##vso" -> "..vso" is the only one that matters.

Reply:
OK, I am happy not being a culprit then. 👼